### PR TITLE
Add build test lint action

### PR DESCRIPTION
**Pull request by mittonface**
Probably worth noting that this is failing at the moment. But this is an instance of the workflow doing its job.

https://github.com/mittonface/VaxFinder-Frontend/runs/2476308686?check_suite_focus=true

### CODE REVIEW

File: .github/workflows/build-test-lint.yml [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+      - name: Install
+        run: yarn install
+
+      - name: Build
+        run: yarn build
+
+      - name: Test
+        run: yarn test --watchAll=false 
+
+      - name: Lint
```
Comment (chardielxr) [inline .github/workflows/build-test-lint.yml:25]: Probably doesn't matter for the size of the codebase, but I generally move lint to the top. Linting fails relatively fast, I have test suites that can take over 15 minutes to run :)

Comment (chardielxr) [inline .github/workflows/build-test-lint.yml:25]: Probably doesn't matter for the size of the codebase, but I generally move lint to the top. Linting fails relatively fast, I have test suites that can take over 15 minutes to run :)

Comment (mittonface) [reply .github/workflows/build-test-lint.yml:25]: This is low-key genius! I've never thought of this.

File: .github/workflows/build-test-lint.yml [hunk 2]
```diff
@@ -0,0 +1,28 @@
+
```
Comment (chrisahardie) [inline .github/workflows/build-test-lint.yml:1]: I'm not familiar with actions, but should we be setting the CI env var to true? It's a matter of course for all my CI scripts.

Comment (patcon) [reply .github/workflows/build-test-lint.yml:1]: Good call, but I believe GH Actions already sets this as default, so no need to specify: https://docs.github.com/en/actions/reference/environment-variables

Good eyes though 👀

**Comment by chrisahardie** [comment]
@mittonface Wanna pull from `main` and see if your shiny new action will give us a greenlight?

**Review by chrisahardie — APPROVED** [review]

**Comment by chrisahardie** [comment]
Thanks @mittonface , I believe this is working a treat! https://github.com/mittonface/VaxFinder-Frontend/actions/runs/800022702

**Comment by patcon** [comment]
Yes, I concur: totally a treat, indeed 🍬

**Pull request merged**